ledger/complete/mtrie/forest_test.go

```diff
@@ -1072,3 +1068,38 @@ func TestNow(t *testing.T) {
 	require.Equal(t, updatedRoot, updatedRoot2)
 	require.Equal(t, 2, size)
 }
+
+func TestPurgeCacheExcept(t *testing.T) {
```

Comment:
Can we also add test cases for purging empty cache, purging except the trie which does not exist inside, purge with only the purged trie in

Reply:
added more tests at [c1dc91f](https://github.com/onflow/flow-go/pull/2893/commits/c1dc91f304eb6009b13757e256b90e3bc7470d3b).